server/api/scheduler.go

```diff
@@ -81,48 +85,21 @@ func (h *schedulerHandler) CreateScheduler(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	switch name {
-	case schedulers.BalanceLeaderName:
-		if err := h.AddBalanceLeaderScheduler(); err != nil {
-			h.r.JSON(w, http.StatusInternalServerError, err.Error())
-			return
-		}
-	case schedulers.BalanceWitnessName:
-		if err := h.AddBalanceWitnessScheduler(); err != nil {
-			h.r.JSON(w, http.StatusInternalServerError, err.Error())
-			return
-		}
-	case schedulers.TransferWitnessLeaderName:
-		if err := h.AddTransferWitnessLeaderScheduler(); err != nil {
-			h.r.JSON(w, http.StatusInternalServerError, err.Error())
-			return
-		}
-	case schedulers.HotRegionName:
-		if err := h.AddBalanceHotRegionScheduler(); err != nil {
-			h.r.JSON(w, http.StatusInternalServerError, err.Error())
-			return
-		}
-	case schedulers.EvictSlowTrendName:
-		if err := h.AddEvictSlowTrendScheduler(); err != nil {
-			h.r.JSON(w, http.StatusInternalServerError, err.Error())
-			return
-		}
-	case schedulers.BalanceRegionName:
-		if err := h.AddBalanceRegionScheduler(); err != nil {
-			h.r.JSON(w, http.StatusInternalServerError, err.Error())
-			return
-		}
-	case schedulers.LabelName:
-		if err := h.AddLabelScheduler(); err != nil {
-			h.r.JSON(w, http.StatusInternalServerError, err.Error())
-			return
+	tp, ok := types.SchedulerStr2Type[name]
+	if !ok {
+		h.r.JSON(w, http.StatusBadRequest, "unknown scheduler")
+		return
+	}
+	var args []string
+	collector := func(v string) {
+		if args == nil {
+			args = []string{}
```

Comment:
maybe can
```suggestion
```

Reply:
Do you mean this?

```go
        args := []string{}
	collector := func(v string) {
		args = append(args, v)
	}
```

This is incompatible with before because when there are no args, args should be a null pointer, otherwise unit tests will fail.